apps/system/js/notifications.js

```diff
@@ -403,44 +402,57 @@ var NotificationScreen = {
      */
     var dir = (detail.bidi === 'auto' || typeof detail.bidi === 'undefined') ?
       document.documentElement.dir : detail.bidi;
-
+    // As per the UX,in case for sms notification ,the notification title
+    // will be the count along with the 'new message' text,but currently in
+    // 'detail' object, name of the sender is coming. in other cases title text
+    // in 'detail' object is appropriate.
+    var cIndex = this.unReadNotificationIndex(manifestURL);
+    if (cIndex === -1) {
     // We need to animate the ambient indicator when the toast
     // timesout, so we skip updating it here, by passing a skip bool
-    this.addUnreadNotification(detail.id, true);
-
+    this.addUnreadNotification(manifestURL, true);
+    cIndex = this.unReadNotificationIndex(manifestURL);
     var notificationNode = document.createElement('div');
     notificationNode.classList.add('notification');
     notificationNode.setAttribute('role', 'link');
-
+    notificationNode.setAttribute('id', manifestURL);
+    notificationNode.setAttribute('tabindex', '0');
     notificationNode.dataset.notificationId = detail.id;
     notificationNode.dataset.noClear = behavior.noclear ? 'true' : 'false';
     notificationNode.lang = detail.lang;
     notificationNode.dataset.predefinedDir = detail.bidi;
-
     notificationNode.dataset.obsoleteAPI = 'false';
     if (typeof detail.id === 'string' &&
-        detail.id.indexOf('app-notif-') === 0) {
+      detail.id.indexOf('app-notif-') === 0) {
       notificationNode.dataset.obsoleteAPI = 'true';
     }
     var type = detail.type || 'desktop-notification';
     notificationNode.dataset.type = type;
     notificationNode.dataset.manifestURL = manifestURL;
-
-    if (detail.icon) {
+    if (detail.appIcon) {
       var icon = document.createElement('img');
-      icon.src = detail.icon;
+      icon.src = detail.appIcon;
```

Comment:
for missed call scenario this change will result in wrong icon displayed, see attached screenshot: 
![screenshot from 2015-11-07 19 31 50](https://cloud.githubusercontent.com/assets/7557901/11016076/ce76c822-8586-11e5-84d9-25c3bcda3905.png)
it happens because ``detail.appIcon == "app://communications.gaiamobile.org/style/icons/communications_45.png"`` while ``detail.icon == "app://communications.gaiamobile.org/dialer/style/icons/dialer_84.png"``